NON_CONTAINERIZED_DEV_README.md

```diff
@@ -1,49 +1,101 @@
+5. Run `bundler -v` to ensure Bundler is installed.  If not, `gem install bundler`.
+6. Run `node -v` to ensure Node is installed. If not, install via https://nodejs.org/en/download/
+7. Run `yarn -v` to ensure Yarn is installed. If not, install via `brew install yarn`
+8. `cd` to where you have the `single_cell_portal_core` Git repo checked out.
+9. Run `bundle install`
+  - That might fail with a message that contains "An error occurred while installing bson_ext"
+  - If so, run: ``gem install bson_ext -v '1.5.1' --source 'https://rubygems.org/' -- --with-cflags="-Wno-error=implicit-function-declaration"``
+  - If that fails, follow steps from https://stackoverflow.com/a/64248633, i.e.:
+    - Run `cd /Users/$(whoami)/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/bson_ext-1.5.1/ext/cbson`
+    - Insert one line, near to the top of the file, `bson_buffer.h`:
 
-1. Run `ruby -v` to ensure Ruby 3.1.2 is installed on your local machine.  If not, [install rbenv](https://github.com/rbenv/rbenv#installation) 
-(via `brew install rbenv` if on macOS) then `rbenv init` to set up rbenv in your shell. Then close out terminal and reopen 
-and run `rbenv install 3.1.2`. (version is current as of 08/09/2022)
-2. Run `bundler -v` to ensure Bundler is installed.  If not, `gem install bundler`.
-3. Run `node -v` to ensure Node is installed. If not, install via https://nodejs.org/en/download/
-4. Run `yarn -v` to ensure Yarn is installed. If not, install via `brew install yarn`
-5. `cd` to where you have the `single_cell_portal_core` Git repo checked out.
-6. Run `bundle install`
-7. Run `yarn install`
-8. Run `./rails_local_setup.rb` to will write out required variables into an shell env file (using your Broad username 
+```
+/* A buffer */
+typedef struct bson_buffer* bson_buffer_t;
+/* A position in the buffer */
+typedef int bson_buffer_position;
+
+/***** THE FOLLOWING IS THE LINE YOU NEED TO INSERT ****/
+int bson_buffer_get_max_size(bson_buffer_t buffer); 
+
+ /* Allocate and return a new buffer.
+ * Return NULL on allocation failure. */
+bson_buffer_t bson_buffer_new(void);
+```
+  -
+    - Run `pwd`, confirm you are still in `/Users/$(whoami)/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/bson_ext-1.5.1/ext/cbson`
+    - Run `make`, which will output warnings you can ignore
+    - Run `cd ../..`, i.e., go to the `bson_ext-1.5.1` directory
+    - Run `gem spec ../../cache/bson_ext-1.5.1.gem --ruby > ../../specifications/bson_ext-1.5.1.gemspec`
+    - Run `gem list bson_ext`, and confirm it outputs:
+
+```
+*** LOCAL GEMS ***
+bson_ext (1.5.1)
+```
+  -
+    - Change directories back to `single_cell_portal_core`
+10. Run `yarn install`
+11. Run `./rails_local_setup.rb` to will write out required variables into an shell env file (using your Broad username 
```

Comment:
For my new M1, I had to go searching for info on how to set up Vault. Here's the link to the DevOps doc:
https://docs.google.com/document/d/11pZE-GqeZFeSOG0UpGg_xyTDQpgBRfr0MLxpxvvQgEw/edit#heading=h.trehvi2yr632
For onboarding, do steps 5-8. For migrating to a new laptop, skip step 7 and copy over your GitHub Personal Access Token.